### DISCUSSION TITLE
How to find pixel_per_meter in Distance calculation

### DISCUSSION THREAD

**Discussion by mr-mainak** [ultralytics/ultralytics#8706] [Q&A]
Hi, greetings of the day.
Can anybody please give reference to how to find pixel_per_meter? In here I see it's set to 10 as default. How to find this for different RTSP linked camera? any help is highly appreciated.
Regard

**Comment by pderrenger** [comment]
Hello! 👋

The `pixel_per_meter` value is essentially a conversion factor that translates pixels in the image to real-world units (meters in this case). This value can vary based on camera setup, focal length, distance from the object, and the actual dimensions of the object in view.

To find this value for your specific RTSP linked camera, you would typically need to calibrate your camera using known measurements in the scene. Here's a simple approach:

1. Place an object with a known size (in meters) within the camera's view.
2. Measure the size of the object in pixels directly from the image.
3. Calculate `pixel_per_meter` as `number_of_pixels / real_world_size_in_meters`.

For example, if a 1-meter long object takes up 100 pixels in your image, your `pixel_per_meter` would be `100 / 1 = 100`.

Keep in mind, this is a simplified method and assumes a flat, frontal view. For more accurate calibration, especially in 3D space or varying depths, you might need to look into more complex camera calibration methods.

Hope this helps! Let us know if you have any more questions. 😊

**Reply by mr-mainak** [reply]
@pderrenger Thank you

**Reply by mr-mainak** [reply]
@pderrenger suppose the camera is giving top view like a traffic camera. In that case there is a depth element. How can one calculate the PPM in that case? Can you suggest some approach.
Thanks

**Reply by pderrenger** [reply]
Hello!

For a top-view camera setup where depth plays a role, calculating the pixel_per_meter (PPM) can be a bit more complex due to perspective changes. One approach is to use a calibration pattern (like a checkerboard) placed at different depths in the camera's field of view. Here’s a simplified method:

1. Place the calibration pattern at known distances (depths) from the camera.
2. Measure the pattern size in pixels at each distance.
3. Calculate PPM for each distance as `pixels / real_world_size`.

This will give you a PPM value that varies with depth, which you can use to interpolate PPM for any given depth in your application.

For more precise calibration, consider using computer vision techniques or specialized software that can account for camera distortion and perspective.

Hope this helps! Let me know if you need further assistance. 😊
